ferrocene/ci/scripts/reset-mtime-to-last-commit.sh

```diff
@@ -38,7 +46,12 @@ for repo in . $(git config --file .gitmodules --get-regexp path | awk '{ print $
     #
     # `--` instructs touch to treat all following arguments as files, even if
     # their names start with `-` (and would thus be interpreted as an option).
-    git ls-tree HEAD -rz --name-only | xargs -0 touch -c -d "${last_commit_date}" --
+    if [[ "${OSTYPE}" =~ ^darwin.* ]]; then
+        # Darwin's bash does not support RFC2822 on `touch`, use a UNIX timestamp
+        git ls-tree HEAD -rz --name-only | xargs -0 touch -c -t "${LAST_COMMIT_DATE}" --
```

Comment:
We should probably just always use UNIX timestamp?

Anyways, on Mac:

```
bash-3.2$ cd git/ferrocene/ferrocene/
bash-3.2$ LAST_COMMIT_DATE="$(git log -1 "--format=%ct" HEAD)"
bash-3.2$ echo $LAST_COMMIT_DATE
1712077253
bash-3.2$ touch -c -t "${LAST_COMMIT_DATE}" foo
```

Failing:
```
bash-3.2$ LAST_COMMIT_DATE="$(git log -1 "--format=%cD" HEAD)"
bash-3.2$ echo $LAST_COMMIT_DATE
Tue, 2 Apr 2024 10:00:53 -0700
bash-3.2$ touch -c -d "${LAST_COMMIT_DATE}" foo
touch: out of range or illegal time specification: YYYY-MM-DDThh:mm:SS[.frac][tz]
```

Reply:
We cannot do that since Linux won't take unix timestamps.